src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java

```diff
@@ -492,10 +492,12 @@ public final void setSortImportsInGroupAlphabetically(boolean value) {
      *        user value.
      */
     public final void setCustomImportOrderRules(final String inputCustomImportOrder) {
-        for (String currentState : GROUP_SEPARATOR_PATTERN.split(inputCustomImportOrder)) {
-            addRulesToList(currentState);
+        if (!inputCustomImportOrder.isEmpty()) {
```

Comment:
do we need this condition ?
what is side effect if we do not have it.

Reply:
This is from your https://github.com/checkstyle/checkstyle/issues/7128#issuecomment-544254686.
> Setter need to be updated to expect empty string and keep empty list for `customImportOrderRules` class field.

There _is_ a difference with and without the condition. Since we say that `""` is the default value, suppose the user actually sets the value explicitly to `""`:
```
D:\checkstyletest>type config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://checkstyle.sourceforge.net/dtds/configuration_1_3.dtd">

<module name = "Checker">
    <module name="TreeWalker">
        <module name="CustomImportOrder">
            <property name="customImportOrderRules" value=""/>
        </module>
    </module>
</module>
```

On the current release, this causes Checkstyle to crash:
```
D:\checkstyletest>java -jar -Duser.language=en -Duser.country=US checkstyle-8.30-all.jar -c config.xml test\*.java
com.puppycrawl.tools.checkstyle.api.CheckstyleException: cannot initialize module TreeWalker - cannot initialize module CustomImportOrder - Cannot set property 'customImportOrderRules' to ''
        at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:473)
        at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:198)
        at com.puppycrawl.tools.checkstyle.Main.runCheckstyle(Main.java:396)
        at com.puppycrawl.tools.checkstyle.Main.runCli(Main.java:324)
        at com.puppycrawl.tools.checkstyle.Main.execute(Main.java:187)
        at com.puppycrawl.tools.checkstyle.Main.main(Main.java:124)
Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: cannot initialize module CustomImportOrder - Cannot set property 'customImportOrderRules' to ''
        at com.puppycrawl.tools.checkstyle.TreeWalker.setupChild(TreeWalker.java:135)
        at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:198)
        at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:468)
        ... 5 more
Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: Cannot set property 'customImportOrderRules' to ''
        at com.puppycrawl.tools.checkstyle.api.AutomaticBean.tryCopyProperty(AutomaticBean.java:237)
        at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:191)
        at com.puppycrawl.tools.checkstyle.TreeWalker.setupChild(TreeWalker.java:130)
        ... 7 more
Caused by: java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.apache.commons.beanutils.PropertyUtilsBean.invokeMethod(PropertyUtilsBean.java:2128)
        at org.apache.commons.beanutils.PropertyUtilsBean.setSimpleProperty(PropertyUtilsBean.java:2109)
        at org.apache.commons.beanutils.BeanUtilsBean.copyProperty(BeanUtilsBean.java:437)
        at com.puppycrawl.tools.checkstyle.api.AutomaticBean.tryCopyProperty(AutomaticBean.java:227)
        ... 9 more
Caused by: java.lang.IllegalStateException: Unexpected rule:
        at com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck.addRulesToList(CustomImportOrderCheck.java:913)
        at com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck.setCustomImportOrderRules(CustomImportOrderCheck.java:496)
        ... 17 more
Checkstyle ends with 1 errors.
```
With the additional condition, we can prevent that from happening:
```
D:\checkstyletest>type config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://checkstyle.sourceforge.net/dtds/configuration_1_3.dtd">

<module name = "Checker">
    <module name="TreeWalker">
        <module name="CustomImportOrder">
            <property name="customImportOrderRules" value=""/>
        </module>
    </module>
</module>

D:\checkstyletest>type test\*.java

test\Test.java


import java.util.Scanner;

public class Test {

}

D:\checkstyletest>java -jar -Duser.language=en -Duser.country=US checkstyle-8.31-SNAPSHOT-all.jar -c config.xml test\*.java
Starting audit...
Audit done.
```

Reply:
thanks a lot for explanation.